### PR TITLE
chore: remove `eth-phishing-detect`

### PR DESCRIPTION
**Pull request by mikesposito**
## Explanation

After porting `PhishingDetector` into `@metamask/phishing-controller`, `eth-phishing-detect` is now unused and can be removed.

### `@metamask/phishing-controller`

_No consumer changes_

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Comment by cryptodev-2s** [comment]
can we remove also the directory `types/eth-phishing-detect`

**Comment by mikesposito** [comment]
> can we remove also the directory `types/eth-phishing-detect`

Absolutely yes, good catch! done in 4086987

File: yarn.lock [hunk 1]
```diff
@@ -3374,7 +3374,6 @@ __metadata:
     "@types/jest": "npm:^27.4.1"
     "@types/punycode": "npm:^2.1.0"
     deepmerge: "npm:^4.2.2"
-    eth-phishing-detect: "npm:^1.2.0"
```
Comment (mikesposito) [inline yarn.lock:3377]: This is still in the tree due to this: https://github.com/MetaMask/metamask-mobile/pull/11111#discussion_r1752309640

**Review by cryptodev-2s — APPROVED** [review]

**Review by legobeat — DISMISSED** [review]

File: yarn.lock (hunk 1 above)
Comment (legobeat) [reply yarn.lock:3377]: Still true today? I guess it can be removed in a follow-up in any case.

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**